internal/service/fw/api_access_codes_resource.go

```diff
@@ -150,15 +152,24 @@ func (r *AccessCodesResource) Delete(ctx context.Context, req resource.DeleteReq
 		return
 	}
 
-	httpRes, err := r.client.FWAPI.
-		AccessCodesAPI.
-		DeleteSingleAccessCodes(ctx, data.AccessKey.ValueString()).
-		Execute()
-	if err != nil {
-		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
-			return
+	err := retry.RetryContext(ctx, InternalDomainListOperationTimeout, func() *retry.RetryError {
```

Comment:
I would keep a separate timeout constant

```suggestion
	err := retry.RetryContext(ctx, AccessCodeOperationTimeout, func() *retry.RetryError {
```

Reply:
Fixed